src/hooks/use-search-categories.tsx

```diff
@@ -0,0 +1,88 @@
+import React, { ReactNode, useEffect, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { useRecoilValue, useSetRecoilState } from 'recoil';
+import { DropDownOption } from '@/components/drop-down-options/drop-down-options';
+import { Deck } from '@/models/deck';
+import { paths } from '@/routing/paths';
+import { userDecksSortedState, userDecksState } from '@/state/user-decks';
+import { useDecksClient } from './api/use-decks-client';
+import { useSearchResultFilter } from './use-search-result-filter';
+
+const ALL_SEARCH_CATEGORIES = ['my decks', 'public decks', 'users'] as const;
+type SearchCategory = typeof ALL_SEARCH_CATEGORIES[number];
+
+export const useSearchCategories = (isCaseSensitive: boolean) => {
+  const [searchCategory, setSearchCategory] = useState(getSearchCategories()[0]);
+  const [searchValue, setSearchValue] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+
+  const { filterSearchResults } = useSearchResultFilter(isCaseSensitive);
+
+  const decksClient = useDecksClient();
+  const setUserDecks = useSetRecoilState(userDecksState); // global user decks
+  const decks = useRecoilValue(userDecksSortedState);
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    fetchResources();
+  }, []);
+
+  const searchOptions = getSearchOptions();
+  const searchResults = filterSearchResults(searchOptions[searchCategory.id], searchValue);
+
+  return {
+    searchValue,
+    searchResults,
+    searchCategory,
+    searchCategories: getSearchCategories(),
+    isLoading,
+    setSearchValue,
+    setSearchCategory,
+    navigateToResult: navigateToResult,
+  };
+
+  function navigateToResult(searchResult: DropDownOption<string, ReactNode>) {
+    if (searchCategory.id === 'my decks') {
+      navigate(`${paths.deck}/${searchResult.id}`);
+    }
+  }
+
+  async function fetchResources() {
+    setIsLoading(true);
+    const promises = [];
+    if (decks.length === 0) {
+      promises.push(fetchDecksAndRefresh());
+    }
+    // TODO: Add the other categories
+    await Promise.all([promises]);
+    setIsLoading(false);
+  }
+
+  async function fetchDecksAndRefresh() {
```

Comment:
if we load `/decks` then we would fetch twice right?
i guess it's not a problem but would it make sense to fetch **only** if `userDecks` (the recoil value) is undefined?

i'm not sure about any issues that would cause but it's something to think about

Reply:
I am checking if the `decks` is an empty array where I call this function, so I can conditionally add the promise to the list of fetch resource promises. 

The default value for `decks` is `[]`, I tried switching it to undefined, but I was running into problems. I was planning on fixing this, and shifting around the control flow a bit in a separate MR where I will also add fetching for users and public decks.

Reply:
you're right, this wouldn't work since we'd have no way of knowing if a request is already fired

it'd probably include something regarding exposing whether a request is already occurring and preventing it...

so let's just worry about this later if it ever comes up